keras_cv/models/segmentation/segformer/segformer_aliases.py

```diff
@@ -0,0 +1,244 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+
+from keras_cv.models.segmentation.segformer.segformer import SegFormer
+from keras_cv.models.segmentation.segformer.segformer_presets import presets
+from keras_cv.utils.python_utils import classproperty
+
+ALIAS_DOCSTRING = """SegFormer model.
+
+    For transfer learning use cases, make sure to read the
+    [guide to transfer learning & fine-tuning](https://keras.io/guides/transfer_learning/).
+
+    Args:
+        backbone: a KerasCV backbone for feature extraction.
+        num_classes: the number of classes for segmentation, including the background class.
+
+    Examples:
+    ```python
+    input_data = tf.ones(shape=(8, 224, 224, 3))
+
+    # Randomly initialized backbone
+    backbone = keras_cv.models.MiTBackbone.from_preset("mit_b0_imagenet")
+    segformer = keras_cv.models.SegFormer(backbone=backbone, num_classes=19)
+    output = model(input_data)
+    ```
+"""  # noqa: E501
+
+
+class SegFormerB0(SegFormer):
+    def __new__(
+        cls,
+        num_classes=19,
```

Comment:
We shouldn't specify a default for `num_classes` as it needs to be user-specified. A silent default could be very confusing.

Reply:
Removed. Should it be requested as a mandatory arg?

Reply:
Yes I think this should be required at init time.